src/main/java/com/ace/alfox/lib/Database.java

```diff
@@ -1,56 +1,34 @@
 package com.ace.alfox.lib;
 
-import com.ace.alfox.game.interfaces.IAction;
-import org.iq80.leveldb.DB;
-import org.iq80.leveldb.Options;
-import org.springframework.beans.factory.config.BeanDefinition;
-import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
-import org.springframework.core.type.filter.AnnotationTypeFilter;
+import com.ace.alfox.game.models.Location;
+import com.ace.alfox.game.models.Player;
+import org.dizitart.no2.Nitrite;
+import org.dizitart.no2.NitriteBuilder;
+import org.dizitart.no2.objects.ObjectRepository;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
-import static org.fusesource.leveldbjni.JniDBFactory.*;
-
-import java.io.File;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
+import javax.annotation.PreDestroy;
 
+@Service
 public class Database {
-    /**
-     * Mappings of /perform/{action} to an IAction. Populated in the AlfoxApplication class.
-     */
-    public static Map<String, Class<IAction>> actions = new HashMap<>();
-    private static DB instance;
+    final Nitrite db;
 
-    private Database() {}
+    public ObjectRepository<Player> players;
+    public ObjectRepository<Location> locations;
 
-    public static void findActions(String namespace) throws ClassNotFoundException {
-        ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(true);
-        scanner.addIncludeFilter(new AnnotationTypeFilter(PlayerAction.class));
-        for (BeanDefinition bd : scanner.findCandidateComponents(namespace)) {
-            // use my annotation to get simple name
-            String alias = Class.forName(bd.getBeanClassName()).getAnnotation(PlayerAction.class).alias();
-            System.out.println(alias);
-            Class cl = Class.forName(bd.getBeanClassName());
-            Database.actions.put(alias, cl);
+    public Database(@Value(value = "game") String name) {

```

Comment:
I think this could use some comments. I wasn't really sure what was going on until I saw the tests and then I was like AHA.

Reply:
Fair